rust/ql/lib/codeql/rust/internal/PathResolution.qll

```diff
@@ -1462,10 +1556,16 @@ private ItemNode getOuterScope(ItemNode i) {
 pragma[nomagic]
 private predicate unqualifiedPathLookup(ItemNode ancestor, string name, Namespace ns, ItemNode encl) {
   // lookup in the immediately enclosing item
-  any(RelevantPath p).isUnqualified(name, encl) and
-  ancestor = encl and
-  exists(ns) and
-  not name = ["crate", "$crate", "super", "self"]
+  exists(RelevantPath path |
+    path.isUnqualified(name, encl) and
+    ancestor = encl and
+    exists(ns) and
+    not name = ["crate", "$crate", "super", "self"]
+  |
+    pathUsesNamespace(path, ns)
+    or
+    not pathUsesNamespace(path, _)
```

Comment:
We could add the `exists(ns)` here for clarity, but it doesn't change the semantics.
```suggestion
    not pathUsesNamespace(path, _) and exists(ns)
```